src/api/helpers.jl

```diff
@@ -30,6 +30,27 @@ function h5f_get_dset_no_attrs_hint(file_id)::Bool
     return minimize[]
 end
 
+"""
+    h5f_get_file_image(file_id)
+
+Return a `Vector{UInt8}` containing the file image. Does not include the user block.
+"""
+function h5f_get_file_image(file_id)
+    buffer_length = h5f_get_file_image(file_id, C_NULL, 0)
+    buffer = Vector{UInt8}(undef, buffer_length)
+    h5f_get_file_image(file_id, buffer, buffer_length)
+    return buffer
+end
```

Comment:
Perhaps should be a different function since it has a different return value.

Reply:
I have to look back through, but I think this is pretty common to have a helper return a different value.

Reply:
Did you mean that `h5f_get_file_image(file_id)` should have a different name than `h5f_get_file_image(file_id, buffer::Vector{UInt8})` or `h5f_get_file_image(file_id, buffer, buffer_length)`?

Reply:
That applies to most of the methods in helpers.jl though, does it not?

* The 3-arg method of `h5_get_libversion` returns `0` or throws. The 0-arg helper function returns a `VersionNumber.
* The 1-arg method of `h5_is_library_threadsafe` returns `0` or throws. The 0-arg helper function returns a `Bool`.
* The 3-arg method of `h5a_get_name` returns `0` or throws. The 1-arg helper function returns a `String`.

I believe I'm following page precedent for a function that contains `get` in the name. We have the original C-function as a method but also a Julia friendly helper method that returns what you are getting. Am I misunderstanding your thought here?

Reply:
I guess that's true.